src/nginx/module.cc

```diff
@@ -980,13 +967,12 @@ void ngx_http_esp_exit_timer_event_handler(ngx_event_t *ev) {
 
   ngx_cycle_t *cycle = reinterpret_cast<ngx_cycle_t *>(ev->data);
 
-  if (ev->timedout) {
-    // Timer timed out. This should be exceedingly rare (the shutdown timeout is
-    // very long). If it does happen, reschedule the same timeout again.
+  if (!ngx_exiting && !ngx_terminate) {
```

Comment:
If you're testing `ngx_terminate`, then you could probably test `ngx_quit` here as well.